hack/test-cover.sh

```diff
@@ -15,8 +15,16 @@
 # limitations under the License.
 set -e
 
+TEST_BIN_DIR="$(dirname "${0}")/../dev/testbin"
```

Comment:
What's the motivation behind the test-related changes apart from bumping the k8s version?

Reply:
It was not up to date with https://github.com/kubernetes-sigs/controller-runtime/blob/master/hack/setup-envtest.sh 

Also, it's better to have it in a local directory than updating system binaries (as it was doing before)

Reply:
If you want to keep it in sync, what do you think about downloading it in the scope of `make revendor`?

Reply:
@mvladev